cpp/models/abm/world.cpp

```diff
@@ -85,14 +86,18 @@ void World::migration(TimePoint t, TimeSpan dt)
             auto target_type       = rule(personal_rng, *person, t, dt, parameters);
             auto& target_location  = find_location(target_type, *person);
             auto& current_location = person->get_location();
-            if (m_testing_strategy.run_strategy(personal_rng, *person, target_location, t)) {
-                if (target_location != current_location &&
-                    target_location.get_number_persons() < target_location.get_capacity().persons) {
-                    bool wears_mask = person->apply_mask_intervention(personal_rng, target_location);
-                    if (wears_mask) {
+            person->is_apply_mask_intervention(personal_rng, target_location);
+            // Check if the capacity of targeted Location is not reached
+            if (target_location != current_location &&
+                target_location.get_number_persons() < target_location.get_capacity().persons) {
+                // Check if the Person wears mask if required
+                if ((target_location.is_mask_required() && person->is_wear_mask()) ||
+                    !target_location.is_mask_required()) {
+                    // Perform TestingStrategy if required
+                    if (m_testing_strategy.run_strategy(personal_rng, *person, target_location, t)) {
```

Comment:
Why did the order of if-clauses change?

Also, why not use the return value of "person->is_apply_mask_intervention(personal_rng, target_location)" instead of "person->is_wear_mask()"?

Reply:
After the benchmark run, I can see it does not affect much the runtime (maybe a bit slower in case of checking the capacity first). Hence, I will leave the order the same as before:

- Check if the Person wear mask first:
Running ./bin/abm_benchmark
Run on (8 X 24.1206 MHz CPU s)
CPU Caches:
  L1 Data 64 KiB (x8)
  L1 Instruction 128 KiB (x8)
  L2 Unified 4096 KiB (x2)
Load Average: 5.59, 4.39, 3.86
---------------------------------------------------------------------------
Benchmark                                 Time             CPU   Iterations
---------------------------------------------------------------------------
abm_benchmark/abm_benchmark_50k        4948 ms         4862 ms            1
abm_benchmark/abm_benchmark_100k       9876 ms         9871 ms            1
abm_benchmark/abm_benchmark_200k      19790 ms        19778 ms            1

- Check if the Location's capacity is reached first:
Running ./bin/abm_benchmark
Run on (8 X 24.0182 MHz CPU s)
CPU Caches:
  L1 Data 64 KiB (x8)
  L1 Instruction 128 KiB (x8)
  L2 Unified 4096 KiB (x2)
Load Average: 4.98, 4.80, 4.15
---------------------------------------------------------------------------
Benchmark                                 Time             CPU   Iterations
---------------------------------------------------------------------------
abm_benchmark/abm_benchmark_50k        4871 ms         4836 ms            1
abm_benchmark/abm_benchmark_100k       9931 ms         9887 ms            1
abm_benchmark/abm_benchmark_200k      20216 ms        19953 ms            1